notebooks/304-pytorch-post-training-quantization/304-pytorch-post-training-quantization.ipynb

```diff
@@ -0,0 +1,698 @@
+    "    # update the last FC layer for Tiny ImageNet number of classes\n",
+    "    NUM_CLASSES = 200\n",
+    "    model.fc = nn.Linear(in_features=2048, out_features=NUM_CLASSES, bias=True)\n",
+    "    model.to(device)\n",
+    "    if model_path.exists():\n",
+    "        checkpoint = torch.load(str(model_path), map_location='cpu')\n",
+    "        model.load_state_dict(checkpoint['state_dict'], strict=True)\n",
+    "    else:\n",
+    "        raise RuntimeEror('There is no checkpoint to load')\n",
+    "    return model\n",
+    "\n",
+    "\n",
+    "model = create_model(MODEL_DIR / fp32_checkpoint_filename)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4291ed2f",
+   "metadata": {},
+   "source": [
+    "### Create train and validation dataloaders "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "93213778",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_dataloaders(batch_size=128):\n",
+    "    train_dir = DATASET_DIR / 'train'\n",
+    "    val_dir = DATASET_DIR / 'val' / 'images'\n",
+    "    normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406],\n",
+    "                                     std=[0.229, 0.224, 0.225])\n",
+    "    train_dataset = datasets.ImageFolder(\n",
+    "        train_dir,\n",
+    "        transforms.Compose([\n",
+    "            transforms.Resize(IMAGE_SIZE),\n",
+    "            transforms.ToTensor(),\n",
+    "            normalize,\n",
+    "        ]))\n",
+    "    val_dataset = datasets.ImageFolder(\n",
+    "        val_dir,\n",
+    "        transforms.Compose([\n",
+    "            transforms.Resize(IMAGE_SIZE),\n",
+    "            transforms.ToTensor(),\n",
+    "            normalize\n",
+    "        ]))\n",
+    "\n",
+    "    train_loader = torch.utils.data.DataLoader(train_dataset, batch_size=batch_size, shuffle=True,\n",
+    "                                               num_workers=4, pin_memory=True, sampler=None)\n",
+    "    \n",
+    "    val_loader = torch.utils.data.DataLoader(val_dataset, batch_size=batch_size, shuffle=False,\n",
+    "                                             num_workers=4, pin_memory=True)\n",
+    "    return train_loader, val_loader\n",
+    "\n",
+    "\n",
+    "train_loader, val_loader = create_dataloaders()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "880f550a",
+   "metadata": {},
+   "source": [
+    "### I. Evaluate the loaded model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "19c6d048",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "acc1 = validate(val_loader, model)\n",
+    "print(f'Test accuracy of FP32 model: {acc1:.3f}')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "21a797fe",
+   "metadata": {},
+   "source": [
+    "### Export the FP32 model to ONNX, which is supported by OpenVINO™ Toolkit, to benchmark it in comparison with the INT8 model."
```

Comment:
Make this paragraph plain-text instead of header-text

Reply:
Done